### PR TITLE
Qual clean dolibarrtriggers

### PR DESCRIPTION
**Pull request by W1W1-M**
# Qual cleaner PHP code for dolibarrtriggers class

- Refactored code to make more readable and efficient
- Added VERSIONS array dictionary constant & depreciated individual VERSION_XXX constants
- Moved property init to constructor so that child classes can follow parent constructor
- Various fixes & optimizations to code & comments

### CODE REVIEW

File: htdocs/core/triggers/dolibarrtriggers.class.php [hunk 1]
```diff
@@ -124,31 +143,29 @@ public function getDesc()
 	 *
 	 * @return string Version of trigger file
 	 */
-	public function getVersion()
+	public function getVersion(): string
```
Comment (eldy) [inline htdocs/core/triggers/dolibarrtriggers.class.php:146]: As the transition from PHP7.0 to PHP7.1+ is not yet complete by a lot of users, for the moment, we must avoid the use of the type in the function
"function(): type"
We will continue to wait a little bit that the nb of users with php 7.0 is lower (A lot of users first upgrade their Dolibarr to then upgrade the PHP according to Dolibarr requirement, so having a Dolibarr not hanging on fatal error with PHP 7.0 is still important to allow migration when this direction is used) .

Comment (W1W1-M) [reply htdocs/core/triggers/dolibarrtriggers.class.php]: Thanks for explanation, I'll keep this in mind.

Comment (W1W1-M) [reply htdocs/core/triggers/dolibarrtriggers.class.php]: For reference https://www.php.net/manual/en/migration70.new-features.php

File: htdocs/core/triggers/dolibarrtriggers.class.php [hunk 2]
```diff
@@ -30,61 +31,76 @@ abstract class DolibarrTriggers
...
 	 * @var string module is dolibarr ready
+	 * @deprecated Use self::VERSIONS
+	 * @see self::VERSIONS
 	 */
 	const VERSION_DOLIBARR = 'dolibarr';
 
+	/**
+	 * @var array dictionary of possible module states
+	 */
+	const VERSIONS = [
```
Comment (eldy) [inline htdocs/core/triggers/dolibarrtriggers.class.php:98]: What is the benefit of using an array of const instead of const directly ?

Comment (W1W1-M) [reply htdocs/core/triggers/dolibarrtriggers.class.php:98]: The benefit is limited to code readability. Grouping the 3 `const VERSION_XXX` consts under one array seemed more convenient and discoverable by developers, it also gives a stronger limit to the scope of values that `$version` can take.

**Pull request merged**